.ci/Jenkinsfile

```diff
@@ -344,7 +345,24 @@ def generateFunctionalTestStep(Map args = [:]){
         error(e.toString())
       } finally {
         junit(allowEmptyResults: true, keepLongStdio: true, testResults: "${BASE_DIR}/outputs/TEST-*.xml")
-          archiveArtifacts allowEmptyArchive: true, artifacts: "${BASE_DIR}/outputs/TEST-*.xml"
+        archiveArtifacts allowEmptyArchive: true, artifacts: "${BASE_DIR}/outputs/TEST-*.xml"
+        tearDown(labels: workerLabels)
+      }
+    }
+  }
+}
+
+/**
+* Tear down the setup for the static workers.
+*/
+def tearDown(Map args = [:]){
```

Comment:
Wdyt about having a specific step in the library for cleaning up Go dependencies?

Reply:
IMO the build system should be the one ensuring the workspace is in good shape. Then the CI should only caring to delete the workspace if required.

Though, it seems `go` has got some opinionated approach and the build system does not provide those tools, based on that, we could create a specific step in the library to help with.

There are some questions regarding:
- the new step will always run within the `Go` environmental context?
- if running as a post declarative step, will it work?

Do you think we just need to raise an issue regarding this improvement?

Reply:
I was/am concerned about having the very same method in different pipelines (Beats and here) and probably in many of the other Go projects that we maintain. 

I agree that the build system should keep the workspace and dependencies in a good shape, but as you know there is nothing as maven/gradle to do so.

Regarding your questions:
- I think the Go environment should be the same, just in case the `clean` method could change between different Go versions
- I've been reading about `go clean`. Do you think it could be a best practice to invoke it before tests run? The same we do `mvn clean test` in the Maven world. If so then we should not think about the post step issue, as the environment would always be right. Even better, we could enforce the clean in the set Go env step 🤔